packages/app/src/app/overmind/namespaces/live/liveMessageOperators.ts

```diff
@@ -64,10 +64,12 @@ export const onUserEntered: Operator<
 
   const user = data.users.find(u => u.id === data.joined_user_id);
 
-  effects.notificationToast.add({
-    message: `${user.username} joined the live session.`,
-    status: NotificationStatus.NOTICE,
-  });
+  if (!state.live.notificationsHidden) {
+    effects.notificationToast.add({
```

Comment:
Actually, we can make `effects.notificationToast` aware of this state.

By going to the `overmind/effects/notificationToast` we can add an `initialize` method. This is just a convention.

Something like:

```ts
let _shouldNotify: () => boolean
export default {
  initialize(shouldNotify) {
    _shouldNotify = shouldNotify
  },
  ...existingMethods
}
```

Now, when Overmind is initialized we wire up our effects. In the case of our toasts effect we can now (overmind/onInitialize.ts):

```ts
export const onInitialize: OnInitialize = async (
  { state, effects, actions },
  overmindInstance
) => {
  effects.notificationsToast.initialize({
    shouldNotify: () => !state.live.isLive || (state.live.isLive && !state.live.notificationsHidden)
  })
  ...
}
```

We do not have to do this now, maybe overkill. But this will put that responsibility at a single point in the codebase, which we could expand later as well :)

Reply:
I like it! 

Creating another pull request after this one because I have no idea how to test the changes 😅

Reply:
Wouldn't it make sense to only hide the live notifications when you press the "silence notifications" option?